src/main/323C.c

```diff
@@ -2116,17 +2129,167 @@ INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_800204CC);
 
 INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_80020580);
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_80020638);
+void func_80020638(struct EngineObj* arg0)
+{
+    if (D_80141BDC[0] == 0) {
+        func_800204CC(D_80141BDC + 3, arg0->unk8);
+        if (controller_state & PADRdown) {
+            func_8001540C(0, 0x22, 0);
+            if (D_80141BDC[3] != 2U) {
+                func_800129F0(8);
+            }
+            arg0->unk1++;
+        }
+    }
+}
+
+void func_800206D0(struct EngineObj* arg0)
+{
+    s8 temp_v1;
+
+    if (D_80141BDC[0] == 0) {
+        switch (D_80141BDF[0]) {
+        case 0:
+            if (D_80171EA8 != 0) {
+                func_800127C8(1);
+                func_80013014();
+                D_80171EA8 = 0;
+            }
+            if (engine_obj.unk5A & PADLleft) {
```

Comment:
Doesn't seem clear that this is a copy of the pad data or not but this is not an issue for now

Reply:
Whoops you're right